### PR TITLE
Feat: multiple confirm for basket

### LINKED ISSUE #1812: Basket. Show confirmation screen for several operations
* Show modal with several operations carousel

### LINKED ISSUE #1815: Basket. Sign button
* Operations not selected state
* Operations selected state
* Run valiadetion on click sign
* show modal if some of operations invalid
* Proceed with valid operations only

### PR DESCRIPTION
**Pull request by Asmadek**
* Select and show confirm for multiple txs in basket
* Validate txs on sign
* Show warning if some of transactions invalid

### CODE REVIEW

File: src/renderer/entities/staking/ui/SelectedValidatorsModal/SelectedValidatorsModal.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const SelectedValidatorsModal = ({ isOpen, validators, onClose }: Props)
     >
       <section>
         <ul className="flex flex-col [overflow-y:overlay]">
-          {validators.map((validator) => (
+          {validators?.map((validator) => (
```
Comment (johnthecat) [inline src/renderer/entities/staking/ui/SelectedValidatorsModal/SelectedValidatorsModal.tsx]: Maybe there should be some empty state for case of empty validator list?

File: src/renderer/features/operations/OperationsConfirm/Nominate/ui/Confirmation.tsx [hunk 2]
```diff
@@ -136,7 +137,7 @@ export const Confirmation = ({ secondaryActionButton, onGoBack }: Props) => {
               onClick={toggleValidators}
             >
               <div className="rounded-[30px] px-1.5 py-[1px] bg-icon-accent">
-                <CaptionText className="text-white">{confirmStore.validators.length}</CaptionText>
+                <CaptionText className="text-white">{confirmStore.validators?.length}</CaptionText>
```
Comment (johnthecat) [inline src/renderer/features/operations/OperationsConfirm/Nominate/ui/Confirmation.tsx]: I think there should be either ternary check around `<CaptionText>` node or some default value, like `confirmStore.validators?.length ?? 0`

File: src/renderer/pages/Basket/model/basket-page-model.ts [hunk 3]
```diff
@@ -22,12 +23,31 @@ import {
...
+  invalid: BasketTransaction[];
+}>();
+const proceedValidationWarning = createEvent<{
+  valid: BasketTransaction[];
+  invalid: BasketTransaction[];
+}>();
+const cancelValidationWarning = createEvent();
 
-const $selectedTxs = createStore<ID[]>([]);
+const $selectedTxs = createStore<Set<number>>(new Set());
```
Comment (johnthecat) [inline src/renderer/pages/Basket/model/basket-page-model.ts]: In this case `Set` is used for filtering uniq values, as I understood it. But for me using `Set` inside effector store look unusual. Also, it seems that logic of filtering in 2 different effects propagates and affects shape of storage. Maybe, you can write some helpers for manipulating array and use them inside those effects?

File: src/renderer/pages/Basket/model/basket-page-model.ts [hunk 4]
```diff
@@ -135,34 +159,169 @@ sample({
...
+sample({
+  clock: signStarted,
+  source: { allTransactions: $basketTransactions, selectedTxs: $selectedTxs, invalidTxs: $invalidTxs },
+  fn: ({ allTransactions, selectedTxs }) => allTransactions.filter((t) => selectedTxs.has(t.id)),
+  target: validateFx,
+});
+
+sample({
+  clock: signContinued,
+  source: { allTransactions: $basketTransactions, selectedTxs: $selectedTxs, invalidTxs: $invalidTxs },
```
Comment (johnthecat) [inline src/renderer/pages/Basket/model/basket-page-model.ts:233]: I think you can `combine()` this values and reuse this store in these samples

**Review by johnthecat — COMMENTED** [review]
I'm newbie, so i've commented some generic second-handed issues, i think you can skip them

File: src/renderer/pages/Basket/ui/Basket.tsx [hunk 5]
```diff
@@ -1,52 +1,89 @@
...
   const { t } = useI18n();
 
   const basketTxs = useUnit(basketPageModel.$basketTransactions);
   const selectedTxs = useUnit(basketPageModel.$selectedTxs);
   const invalidTxs = useUnit(basketPageModel.$invalidTxs);
+  const validTxs = useUnit(basketPageModel.$validTxs);
+  const validatingTxs = useUnit(basketPageModel.$validatingTxs);
+  const validationWarningShown = useUnit(basketPageModel.$validationWarningShown);
+
+  console.log('xcmValid', validTxs);
```
Comment (sokolova-an) [inline src/renderer/pages/Basket/ui/Basket.tsx]: do we need it here?

File: src/renderer/pages/Basket/ui/SignOperations.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+  RestakeConfirmation,
+  TransferConfirm,
+  UnstakeConfirmation,
+  WithdrawConfirmation,
+} from '@features/operations/OperationsConfirm';
+import { TransferTypes, XcmTypes } from '@entities/transaction';
+import { networkModel } from '@/src/renderer/entities/network';
+import { OperationTitle } from '@/src/renderer/entities/chain';
+import { getOperationTitle } from '../lib/operation-title';
+import { cnTw } from '@/src/renderer/shared/lib/utils';
```
Comment (sokolova-an) [inline src/renderer/pages/Basket/ui/SignOperations.tsx]: imports

File: src/renderer/pages/Basket/ui/SignOperations.tsx [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+      [TransactionType.NOMINATE]: () => <NominateConfirmation hideSignButton />,
+      [TransactionType.STAKE_MORE]: () => <BondExtraConfirmation hideSignButton />,
+      [TransactionType.REDEEM]: () => <WithdrawConfirmation hideSignButton />,
+      [TransactionType.RESTAKE]: () => <RestakeConfirmation hideSignButton />,
+      [TransactionType.DESTINATION]: () => <PayeeConfirmation hideSignButton />,
+      [TransactionType.UNSTAKE]: () => <UnstakeConfirmation hideSignButton />,
+    };
+
+    // @ts-ignore
+    return Components[type];
```
Comment (sokolova-an) [inline src/renderer/pages/Basket/ui/SignOperations.tsx:95]: why do we need `ts-ignore` here? We don't use all of the `TransactionType` here or is it something else?

File: src/renderer/pages/Basket/ui/SignOperations.tsx [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+            onClick={previousTx}
+          />
+          <div
+            className={cnTw(
+              'rounded-full font-semibold border border-divider w-[77px] h-[42px]',
+              'text-text-secondary flex items-center justify-center',
+              'shadow-shadow-1',
+            )}
+          >
+            {currentTx + 1}/{transactions.length}
```
Comment (sokolova-an) [inline src/renderer/pages/Basket/ui/SignOperations.tsx]: maybe move it into a variable to avoid calculation here?

**Review by sokolova-an — APPROVED** [review]

**Review by tuul-wq — APPROVED** [review]

File: src/renderer/pages/Basket/ui/SignOperations.tsx (hunk 7 above)
Comment (Asmadek) [reply src/renderer/pages/Basket/ui/SignOperations.tsx:95]: Right, we don't use multisig, proxy and technical txs like chill or batch

**Pull request merged**